src/main/java/src/main/java/kafka/KafkaApplication.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+import src.main.java.EventModel;
+import src.main.java.router.EventRouter;
+
+@SpringBootApplication
+public class KafkaApplication {
+
+    public static void main(String[] args) throws Exception {
+
+        ConfigurableApplicationContext context = SpringApplication.run(KafkaApplication.class, args);
+
+        MessageProducer producer = context.getBean(MessageProducer.class);
+        MessageListener listener = context.getBean(MessageListener.class);
+
+        producer.sendEventModelMessage(new EventModel("eventName","body of event"));
+        listener.eventModelLatch.await(10, TimeUnit.SECONDS);
+        context.close();
+    }
+
+    @Bean
+    public MessageProducer messageProducer() {
+        return new MessageProducer();
+    }
+
+    @Bean
+    public MessageListener messageListener() {
+        return new MessageListener();
+    }
+
+    public static class MessageProducer {
+
+        @Value(value = "${eventModel.topic.name}")
+        private String eventModelTopicName;
+
+        @Autowired
+        private KafkaTemplate<String, String> kafkaTemplate;
+
+        @Autowired
+        private KafkaTemplate<String, EventModel> EventModelkafkaTemplate;
+
+        public void sendMessage(final String message) {
+
+            ListenableFuture<SendResult<String, String>> future = kafkaTemplate.send("eventTopic", message);
+
+            future.addCallback(new ListenableFutureCallback<SendResult<String, String>>() {
+
+                @Override
+                public void onSuccess(SendResult<String, String> result) {
+                    System.out.println("Sent message=[" + message + "] with offset=[" + result.getRecordMetadata().offset() + "]");
+                }
+
+                @Override
+                public void onFailure(Throwable ex) {
+                    System.out.println("Unable to send message=[" + message + "] due to : " + ex.getMessage());
+                }
+            });
+        }
+
+        public void sendEventModelMessage(EventModel msg) {
+            EventModelkafkaTemplate.send(eventModelTopicName, msg);
+
+        }
+
+
+    }
+
+    public static class MessageListener {
+        private CountDownLatch eventModelLatch = new CountDownLatch(1);
+
+        @KafkaListener(topics ="${eventModel.topic.name}",containerFactory = "EventModelkafkaListenerContainerFactory")
+        public void listenWithHeaders(
+                @Payload EventModel message,
+                @Header(KafkaHeaders.RECEIVED_PARTITION_ID) int partition) {
+            JSONObject messageJson= new JSONObject(message);
```

Comment:
```suggestion
            JSONObject messageJson = new JSONObject(message);
```

La suggestion c'est juste pour la propreté du code ;).

Sinon sur le code en lui même, pourquoi refaire un JSON quand tu as déjà une instance de ta classe ?